runtime/near-vm-logic/src/logic.rs

```diff
@@ -609,6 +609,36 @@ impl<'a> VMLogic<'a> {
         Ok(self.context.epoch_height)
     }
 
+    /// Returns the stake of an account, if the account is currently a validator. Otherwise returns 0.
+    ///
+    /// # Cost
+    ///
+    /// `base + memory_write_base + memory_write_size * 16 + utf8_decoding_base + utf8_decoding_byte * account_id_len + validator_stake_base`.
+    pub fn validator_stake(
+        &mut self,
+        account_id_len: u64,
+        account_id_ptr: u64,
+        stake_ptr: u64,
+    ) -> Result<()> {
+        self.gas_counter.pay_base(base)?;
+        let account_id = self.read_and_parse_account_id(account_id_ptr, account_id_len)?;
+        let balance = self.ext.validator_stake(&account_id)?.unwrap_or_default();
+        self.gas_counter.pay_base(validator_stake_base)?;
```

Comment:
The base cost payment should come before invocation of ` let balance = self.ext.validator_stake(&account_id)?.unwrap_or_default();`. Otherwise if gas is underpaid the computation costs caused by the contract will not be actually reflected in the burnt gas. We should always charge before performing an operation.

FYI @frol @olonho